atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTable.java

```diff
@@ -124,8 +126,17 @@ public void checkDatabaseVersion() {
 
     @Override
     public void compactInternally() {
-        if (config.compactIntervalMillis() > 0 && checkIfTableHasNotBeenCompactedForCompactIntervalMillis()) {
-            runCompactOnTable();
+        try {
+            if (compactionSemaphore.tryAcquire()) {
+                if (config.compactIntervalMillis() == 0) {
```

Comment:
I'd prefer `config.compactIntervalMillis() == 0 || (config.compactIntervalMillis() > 0 && checkIfTableHasNotBeenCompactedForCompactIntervalMillis())`. Also consider extracting a method for this block to make it easier to read.

Reply:
Fixed.